src/main/java/oss/fosslight/util/ExcelUtil.java

```diff
@@ -2858,7 +2860,195 @@ public static List<OssMaster> readOssList(MultipartHttpServletRequest req, Strin
-	
+
+	/**Excel 헤더 체크*/
+	public static boolean checkHeaderLicenseColumnValidate(Iterator<Cell> cellIterator) {
+		String[] definedColData = new String[] {"LICENSE NAME", "LICENSE TYPE", "NOTICE", "SOURCE CODE",
+				"SPDX SHORT IDENTIFIER", "NICKNAME", "WEBSITE FOR THE LICENSE",
+				"USER GUIDE", "LICENSE TEXT", "ATTRIBUTION", "COMMENT"};
+		boolean allColumnsExist = true;
+
+		List<String> firstRowColData = new ArrayList<>();
+		while (cellIterator.hasNext()) {
+			Cell cell = (Cell) cellIterator.next();
+			firstRowColData.add(getCellData(cell));
+		}
+
+		for (int cIdx = 0; cIdx < definedColData.length; cIdx++) {
+			String colData = firstRowColData.get(cIdx);
+			if (colData != null && definedColData[cIdx].equals(colData.toUpperCase())){
+				continue;
+			} else {
+				allColumnsExist = false;
+				break;
+			}
+		}
+		return allColumnsExist;
+	}
+
+	/**make Grid data to return UI*/
+	public static List<LicenseMaster> readLicenseList(MultipartHttpServletRequest req, String excelLocalPath) throws InvalidFormatException, IOException {
+		Iterator<String> fileNames = req.getFileNames();
+		List<LicenseMaster> licenseMasterList = new ArrayList<>();
+
+		while (fileNames.hasNext()) {
+
+			MultipartFile multipart = req.getFile(fileNames.next());
+			String fileName = multipart.getOriginalFilename();
+			String[] fileNameArray = StringUtil.split(fileName, File.separator);
+			fileName = fileNameArray[fileNameArray.length - 1];
+			File file = new File( "."+excelLocalPath +fileName);
+			FileUtil.transferTo(multipart, file);
+			HSSFWorkbook wbHSSF = null;
+			XSSFWorkbook wbXSSF = null;
+			String[] ext = StringUtil.split(fileName, ".");
+			String extType = ext[ext.length - 1];
+			String codeExt[] = StringUtil.split(codeMapper.selectExtType("11"), ","); // The codeExt array contains the allowed extension types.
+			int count = 0;
+
+			for (int i = 0; i < codeExt.length; i++) {
+				if (codeExt[i].equalsIgnoreCase(extType)) {
+					count++;
+				}
+			}
+
+			if (count != 1) {
+				return null;
+			} else if ("xls".equals(extType) || "XLS".equals(extType)) {
+				try {
+					wbHSSF = new HSSFWorkbook(new FileInputStream(file));
+					HSSFSheet sheet = wbHSSF.getSheetAt(0);
+					int rowSize = sheet.getPhysicalNumberOfRows();
+					for (int rowIndex = 0; rowIndex < rowSize; rowIndex++) {
+						HSSFRow row = sheet.getRow(rowIndex);
+						if (rowIndex == 0) {
+							//여기서 헤더 점검 필요
+							boolean validHeader = checkHeaderLicenseColumnValidate(row.cellIterator());
+							if (!validHeader){
+								log.info("The order and content of header columns must match.");
+								return null;
+							}
+						} else {
+							LicenseMaster licenseMaster = getLicenseDataByColumn(row.cellIterator());
+							if (licenseMaster != null)
+								licenseMasterList.add(licenseMaster);
+						}
+					}
+				} catch (IOException e) {
+					log.error(e.getMessage(), e);
+				} finally {
+					try {
+						wbHSSF.close();
+					} catch (Exception e) {
+					}

```

Comment:
Is there any reason there is no action?

Reply:
thank you for good review. I had to write log code in there